libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -77,13 +69,24 @@
     // Static member
     private static Map<String, SyncManager> INSTANCES = new HashMap<String, SyncManager>();
 
-    // Members
-    private Set<Long> runningSyncIds = new HashSet<Long>();
-    public final String apiVersion;
+    // Keeping track of active syncs (could be waiting for thread or running on a thread)
+    private Map<Long, SyncTask> activeSyncs = new HashMap<>();
+
+    // Flag set when stop is requested
+    private boolean stopRequested = false;
```

Comment:
Instead of having a `boolean` based system, should we consider using an `enum` for states -> `STOP_REQUESTED`, `STOPPING_IN_PROGRESS`, `STOP_COMPLETE`, `RESUME_REQUESTED`, `RESUME_IN_PROGRESS`, and then finally, `SYNC_COMPLETE`? If we use `RESUME`, we should probably pair it with `PAUSE` for clarity.

Reply:
That sounds good. I'm not sure about about the SYNC_COMPLETE state, because once you resume, new jobs could be submitted, so there might always be sync in flights. I don't know if we want to separately track syncs submitted before the stop from the one submitted after the resume.

Reply:
Should we call it `PAUSE` instead of `STOP`, since we have `RESUME`?

Reply:
I started by calling it `pause` but then it's not a perfect `pause` (I'm restarting the sync and reducing repeated work by leveraging the maxTimeStamp). For SOQL sync down a true pause/resume would leverage the `nextRecordsUrl` of the started SOQL request. The problem with that approach is that that URL is not available for ever, so resuming could fail if it happens too long after the pause. For that reason, I decided to keep it simple and leverage what we already have (maxTimeStamp) with the only change being that maxTimeStamp is updated as the sync runs (instead of simply at the end) if the sync's target returns true in for isSyncDownSortedByLatestModification().

Maybe for symmetry, I should rename `resume` to `restart`?

Reply:
Up to you - I don't feel strongly either way.